apps/storefront/src/pages/QuoteDraft/index.test.tsx

```diff
@@ -1275,6 +1331,340 @@ describe('when the user is a B2B customer', () => {
+        contactInfo: quoteInfo.draftQuoteInfo.contactInfo,
+        taxTotal: '500.00',
+        productList: [
+          {
+            basePrice: '10.00',
+            offeredPrice: '10.00',
+            discount: '0.00',
+            imageUrl: 'url',
+            itemId: generatedUuid,
+            options: [],
+            productId: 4451490883947128,
+            productName: 'Unbranded Rubber Cheese',
+            quantity: 100,
+            sku: 'test',
+            variantId: undefined,
+          },
+        ],
+        referenceNumber: '123',
+        currency: {
+          currencyCode: 'USD',
+          currencyExchangeRate: '1.0000000000',
+          decimalPlaces: 2,
+          decimalToken: '.',
+          location: 'left',
+          thousandsToken: ',',
+          token: '$',
+        },
+        message: 'meow',
+        grandTotal: '1000.00',
+        totalAmount: '1500.00',
+      });
+
+      renderWithProviders(<QuoteDraft setOpenPage={vi.fn()} />, {
+        preloadedState: {
+          ...preloadedState,
+          quoteInfo,
+          global: buildGlobalStateWith({
+            blockPendingQuoteNonPurchasableOOS: { isEnableProduct: false },
+            featureFlags: {
+              'B2B-3318.move_stock_and_backorder_validation_to_backend': true,
+            },
+            quoteSubmissionResponse: {
+              value: '0',
+            },
+          }),
+        },
+      });
+
+      server.use(
+        graphql.mutation('CreateQuote', () =>
+          HttpResponse.json(
+            buildQuoteCreateResponseWith({
+              data: {
+                quoteCreate: { quote: { id: 123, createdAt: '1245' } },
+              },
+            }),
+          ),
+        ),
+        graphql.mutation('DeleteCart', ({ variables }) => HttpResponse.json(deleteCart(variables))),
+        graphql.query('GetQuoteInfoB2B', () => HttpResponse.json(quote)),
+        graphql.query('SearchProducts', () => HttpResponse.json({ data: { productsSearch: [] } })),
+      );
+
+      await waitForElementToBeRemoved(() => screen.queryByText(/loading/i));
+
+      const submitButton = await screen.findByRole('button', { name: /Submit/i });
+      expect(submitButton).toBeInTheDocument();
+
+      await userEvent.click(submitButton);
+
+      expect(window.b2b.callbacks.dispatchEvent).toHaveBeenCalledWith('on-quote-create', quoteData);
+
+      expect(getVariantInfoOOSAndPurchase).not.toHaveBeenCalled();
+      expect(mockedUsedNavigate).toHaveBeenCalled();
+      expect(mockedUsedNavigate).toHaveBeenLastCalledWith(`/quoteDetail/123?date=1245`, {
+        state: { to: 'draft' },
+      });
+    });
+
+    it('renders snackbar error if mutation throws product validation erros', async () => {
+      set(window, 'b2b.callbacks.dispatchEvent', vi.fn().mockReturnValue(true));
+      vi.mock('uuid', () => ({
+        v4: vi.fn(() => '1abc-defg1234'),
+        v1: vi.fn(() => `v1-${Math.random().toString()}`),
+      }));
```

Comment:
Instead of mocking `uuid`, use `expect.stringMatching()` and ignore the actual value 🤷